pkg/segment/results/mresults/seriesresult_test.go

```diff
@@ -210,3 +211,28 @@ func Test_reduceRunningEntries(t *testing.T) {
 	_, err = reduceRunningEntries(entries, segutils.Cardinality, functionConstant)
 	assert.NotNil(t, err)
 }
+
+func Test_applyMathFunctionsToResults(t *testing.T) {
+	result := make(map[string]map[uint32]float64)
+	ts := make(map[uint32]float64)
+	ts[1714880880] = -3
+	ts[1714880881] = 2
+	ts[1714880891] = -1
+
+	metricsResults := &MetricsResult{
+		Results: result,
+	}
+
+	function := structs.Function{MathFunction: segutils.Abs}
+
+	err := metricsResults.ApplyFunctionsToResults(function)
+	assert.Nil(t, err)
+	for _, timeSeries := range metricsResults.Results {
+		for _, val := range timeSeries {
+			if val < 0 {
+				t.Errorf("Expected value to be >= 0, got %v", val)
+			}
+		}
+	}
```

Comment:
It would be better to test the exact values we get, instead of just checking they're positive. We'll definitely want to check exact values for other functions we add